api/src/groups/test_groups.py

```diff
@@ -5,993 +5,816 @@
-    generate_managers(auth_client)
-    create_multiple_groups(auth_client.sqla, 4)
-    create_multiple_people(auth_client.sqla, 4)
-    create_multiple_members(auth_client.sqla, 4)
-
-    members = auth_client.sqla.query(Member).all()
 
-    # WHEN member is changed to active
-    for member in members:
-        resp = auth_client.put(url_for('groups.deactivate_member', member_id=member.id))
-        print(member)
-        # THEN assert group is active
-        assert resp.status_code == 200
-        assert resp.json['active'] == False
+@pytest.mark.smoke
+def test_read_all_attendances(auth_client):
+    # GIVEN an empty database
+    # WHEN we read all attendance
+    create_multiple_meetings(auth_client.sqla, random.randint(3, 6))
+    create_multiple_people(auth_client.sqla, random.randint(3, 6))
+    create_multiple_attendance(auth_client.sqla, fraction=0.75)
+    all_attendance = auth_client.sqla.query(Attendance).all()
+    first_meeting = auth_client.sqla.query(Meeting).first()
+    first_person = auth_client.sqla.query(Person).first()
+    # WHEN read all the peron's id and whih meeting they are at
+    # THEN find the attendance in a meeting wich specific meeting id
+    count = 0
+    for attendance in all_attendance:
+        if attendance.meeting_id == first_meeting.id:
+            count+=1
+    resp = auth_client.get(url_for('groups.read_all_attendances', meeting_id = first_meeting.id))
+    print(len(resp.json))
+    # THEN expect the same number of attendacne
+    assert len(resp.json) == count
+# 
+# @pytest.mark.smoke
+# def test_read_attendance_by_member(auth_client):
+#     pass
+# 
+# @pytest.mark.smoke
+# def test_read_attendance_by_meeting(auth_client):
+#     pass
+# 
+@pytest.mark.smoke
+def test_delete_attendance(auth_client):
+    # GIVEN an empty database
+    # WHEN we add in attendance
+    create_multiple_meetings(auth_client.sqla, random.randint(3, 6))
+    create_multiple_people(auth_client.sqla, random.randint(3, 6))
+    first_meeting = auth_client.sqla.query(Meeting).first()
+    first_person = auth_client.sqla.query(Person).first()
+    # WHEN add attendance with perosn_id and meeting_id
+    auth_client.sqla.add(Attendance(meeting_id = first_meeting.id, person_id = first_person.id))
+    auth_client.sqla.commit()
+    # WHEN delete the first meeting
+    resp = auth_client.delete(url_for('groups.delete_attendance', meeting_id = first_meeting.id, person_id = first_person.id))
+    #THEN expect correct message
+    assert resp.status_code == 204
 
-    resp = auth_client.put(url_for('groups.activate_member', member_id=500))
-    # THEN assert group is not found
+    # WHEN delete the not-exist attendance
+    first_person = auth_client.sqla.query(Person).first()
+    resp = auth_client.delete(url_for('groups.delete_attendance', meeting_id = first_meeting.id, person_id = first_person.id))
+    # THEN expect error message
     assert resp.status_code == 404
 
+# 
+# @pytest.mark.smoke
```

Comment:
Tests with this decorator are called "smoke tests." A smoke test is supposed to be a test that will run quickly to give the developer rapid feedback that the system is basically working. The idea is that you want to be able to run these tests frequently; if they take too long, the developer will tend _not_ to run them. Less often, you'll run _all_ the tests to get comprehensive test results.

So don't necessarily mark tons of tests as smoke tests. We should have a few smoke tests in for each module, but not 10's or 100's of them.

Reply:
Sorry about this, I thought this is just a simple marker for tester code. Will fix this issue soon

Reply:
No worries. Note that this "tagging" method is completely flexible. If we wanted to create other subsets of the tests, we could add new tags that would let us test different subsystems more easily. I haven't gone in this direction yet, because you can also run tests by file name, etc.